src/Lecture2.hs

```diff
@@ -211,8 +280,19 @@ The algorithm of merge sort is the following:
 [1,2,3]
 -}
 mergeSort :: [Int] -> [Int]
-mergeSort = error "TODO"
-
+mergeSort xs
+    | length xs < 2 = xs
```

Comment:
`length` is a slow function for lists and use it twice here. It would be more performant to check this with pattern matching instead: an empty list and a singleton list. In that case, no calls to `length` in the beginning are required 🙂